x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authz/store/NativeRolesStoreTests.java

```diff
@@ -807,6 +809,62 @@ public void testBulkDeleteReservedRole() {
         verify(client, times(0)).bulk(any(BulkRequest.class), any());
     }
 
+    /**
+     * Make sure all top level fields for a RoleDescriptor have default values to make sure they can be set to empty in an upsert
+     * call to the roles API
+     */
+    public void testAllTopFieldsHaveEmptyDefaultsForUpsert() throws IOException, IllegalAccessException {
```

Comment:
The purpose of this test is to catch any new top level role descriptor fields that do not provide defaults since it's easy to forget to add that and there is no guarantee that a regular unit test would catch that.